core/client/fs/src/main/java/alluxio/client/block/BlockStoreClient.java

```diff
@@ -154,8 +154,17 @@ public BlockInStream getInStream(BlockInfo info, InStreamOptions options,
     } catch (UnavailableException e) {
       //When BlockInStream created failed, it will update the passed-in failedWorkers
       //to attempt to avoid reading from this failed worker in next try.
+      LOG.info("Added {} to failedWorkers for {}", dataSource, e.toString());
       failedWorkers.put(dataSource, System.currentTimeMillis());
       throw e;
+    } catch (IOException e) {
+      // Some places convert UnavailableException to IOException,
```

Comment:
could you please point out the place of conversion?

Reply:
Is it possible not to convert that?

Reply:
> could you please point out the place of conversion?

The place of conversion is here: https://github.com/Alluxio/alluxio/blob/master/core/client/fs/src/main/java/alluxio/client/block/stream/BlockWorkerClient.java#L64

Reply:
> Is it possible not to convert that?

It is possible, but it will affect all use cases. I think catching one more exception is ok.